bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinBaseHandler.java

```diff
@@ -0,0 +1,202 @@
+    protected abstract void changeFanSpeed(String fanSpeed) throws DaikinCommunicationException;
+
+    // Power, Temp, Fan and Mode are handled in this base class. Override this to handle additional channels. 
+    protected abstract boolean handleCommandInternal(ChannelUID channelUID, Command command) throws DaikinCommunicationException;
+
+
+    public DaikinBaseHandler(Thing thing, DaikinDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        logger = LoggerFactory.getLogger(getClass());
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        boolean handled = false;
+        try {
+            switch (channelUID.getId()) {
+                case DaikinBindingConstants.CHANNEL_AC_POWER:
+                    if (command instanceof OnOffType) {
+                        changePower(((OnOffType) command).equals(OnOffType.ON));
+                        handled = true;
+                    }
+                    break;
+                case DaikinBindingConstants.CHANNEL_AC_TEMP:
+                    if (changeSetPoint(command)) {
+                        handled = true;
+                    }
+                    break;
+                case DaikinBindingConstants.CHANNEL_AIRBASE_AC_FAN_SPEED:
+                case DaikinBindingConstants.CHANNEL_AC_FAN_SPEED:
+                    if (command instanceof StringType) {
+                        changeFanSpeed(((StringType) command).toString());
+                        handled = true;
+                    }
+                    break;
+                case DaikinBindingConstants.CHANNEL_AC_MODE:
+                    if (command instanceof StringType) {
+                        changeMode(((StringType) command).toString());
+                        handled = true;
+                    }
+                    break;
+            }
+            if (!handled && !handleCommandInternal(channelUID, command)) {
+                logger.debug("Received command ({}) of wrong type for thing '{}' on channel {}", command, thing.getUID().getAsString(), channelUID.getId());
+            }
+        } catch (DaikinCommunicationException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Daikin AC Unit");
+        DaikinConfiguration config = getConfigAs(DaikinConfiguration.class);
+        if (config.host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host address must be set");
+        } else {
+            webTargets = new DaikinWebTargets(config.host);
+            refreshInterval = config.refresh;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
```

Comment:
```suggestion
    public void handleRemoval() {
        stopPoll();
        super.handleRemoval();
    }
```

and the same for dispose. It's usally better to first remove your own stuff and then let the super class clean up it's own.

Reply:
Thanks! This was inherited from the old code. I'll make the change.